### ISSUE TITLE
CAA unparsed plain text string?

### ISSUE THREAD

**Issue by JaneX8** [silverwind/caa#1]
Hi,
Great plugin! :) Any chance that I can return the plain text record as well? In addition to the parsed object. I'd like to use an option or something to acces for example string:

` 0 issuewild "letsencrypt.org" 0 issue "letsencrypt.org" 0 iodef "mailto:john@example.com" ` 

in addition to the object.

**Comment by silverwind** [comment]
The format of the CAA object is defined in [dns-packet](https://github.com/mafintosh/dns-packet), so any change to it should happen there. I'm not sure what the benefit of such a string format would because the data itself is not a string in the DNS encoding. Your example should return:

```` js
[
  {flags: 0, tag: "issuewild", value: "letsencrypt.org"}
  {flags: 0, tag: "issue", value: "letsencrypt.org"}
  {flags: 0, tag: "iodef", value: "mailto:john@example.com"}
]
````

**Comment by silverwind** [comment]
Sorry, but I don't see any value in a string format. It would get pretty messy and hard to parse once one considers the optional [rfc6844bis](https://tools.ietf.org/html/draft-ietf-acme-caa-05) extensions which have data like this:

````
   example.com. IN CAA 0 issue "example.net; \
     accounturi=https://example.net/account/1234; \
     validationmethods=dns-01"
   example.com. IN CAA 0 issue "example.net; \
     accounturi=https://example.net/account/2345; \
     validationmethods=http-01"
````

**Comment by JaneX8** [comment]
Hi @silverwind,
Thanks or the replies. I figured that the way of setting up a CAA record is to type in some text / string into my DNS config (which I believe was a oneliner, containing all info). But in your last commend it seems multiline. 

The result it returns now is already parsed as in nicely showing keys and values in an JSON array. But the original string is in my opinion `0 issue "example.net; accounturi=https://example.net/account/1234; validationmethods=dns-01` (with or without multiline). Right?

**Comment by silverwind** [comment]
No multiline. A CAA record ist just name/value pair and a bitflag byte where name (called tag in the rfc) is for example `issue`. The reason it cannot really be represented in a single string is that it's pretty common to have multiple CAA records and there is no clear way on how those would be concatenated. Even if your DNS provider supports a single string format for multiple records, that method would probably not work for other providers, so I chose the array of objects form which is easier to work with.

Your example is correct, it could be represented like this per record:

````
0 issue "example.net; accounturi=https://example.net/account/1234; validationmethods=dns-01"
````

You could transform the object to this form pretty easily using a template string:

```` js
`${flags} ${tag} "${value}"`
````
